modules/layers/src/contour-layer/marching-squares.js

```diff
@@ -1,43 +1,51 @@
 // All utility mehtods needed to implement Marching Squres algorithm
 // Ref: https://en.wikipedia.org/wiki/Marching_squares
 import assert from 'assert';
+import {log} from '@deck.gl/core';
+import {ISOLINES_CODE_OFFSET_MAP, ISOBANDS_CODE_OFFSET_MAP} from './marching-squares-codes';
 
-// Table to map code to the intersection offsets
-// All offsets are relative to the center of marching cell (which is top right corner of grid-cell)
-const OFFSET = {
-  N: [0, 0.5],
-  E: [0.5, 0],
-  S: [0, -0.5],
-  W: [-0.5, 0]
+export const CONTOUR_TYPE = {
+  ISO_LINES: 1,
+  ISO_BANDS: 2
 };
 
-// Note: above wiki page invertes white/black dots for generating the code, we don't
-const CODE_OFFSET_MAP = {
-  0: [],
-  1: [[OFFSET.W, OFFSET.S]],
-  2: [[OFFSET.S, OFFSET.E]],
-  3: [[OFFSET.W, OFFSET.E]],
-  4: [[OFFSET.N, OFFSET.E]],
-  5: [[OFFSET.W, OFFSET.N], [OFFSET.S, OFFSET.E]],
-  6: [[OFFSET.N, OFFSET.S]],
-  7: [[OFFSET.W, OFFSET.N]],
-  8: [[OFFSET.W, OFFSET.N]],
-  9: [[OFFSET.N, OFFSET.S]],
-  10: [[OFFSET.W, OFFSET.S], [OFFSET.N, OFFSET.E]],
-  11: [[OFFSET.N, OFFSET.E]],
-  12: [[OFFSET.W, OFFSET.E]],
-  13: [[OFFSET.S, OFFSET.E]],
-  14: [[OFFSET.W, OFFSET.S]],
-  15: []
+const Z_OFFSET = 0.002;
+
+const DEFAULT_THRESHOLD_DATA = {
+  zIndex: 0,
+  zOffsetScale: 1
 };
 
+// Utility methods
+
+function getVertexCode({weight, threshold}) {
```

Comment:
How frequently is this called? For private utility functions that are performance-sensitive, you can skip named parameters to avoid the creation and garbage collection of small objects.

Reply:
yeah this is performed for each cell, i.e. called very frequently, I will pass individual parameters, to avoid object creation.